internal/keygen/keygen.go

```diff
@@ -1,25 +1,151 @@
 package keygen
 
 import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/vultisig/mobile-tss-lib/tss"
 	"github.com/vultisig/vultisigner/config"
+	"github.com/vultisig/vultisigner/internal/logging"
 	"github.com/vultisig/vultisigner/internal/types"
-
-	"github.com/vultisig/mobile-tss-lib/coordinator"
+	"github.com/vultisig/vultisigner/relay"
 )
 
-func JoinKeyGeneration(kg *types.KeyGeneration) error {
-	relayServer := config.AppConfig.Relay.Server
+func JoinKeyGeneration(kg *types.KeyGeneration) (string, string, error) {
+	keyFolder := config.AppConfig.Server.VaultsFilePath
+	serverURL := config.AppConfig.Relay.Server
+
+	server := relay.NewServer(serverURL)
+	/*	if err := server.RegisterSession(kg.Session, kg.Key); err != nil {
+		return "", "", fmt.Errorf("failed to register session: %w", err)
+	}*/
+
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	partiesJoined, err := server.WaitForSessionStart(ctx, kg.Session)
+	logging.Logger.WithFields(logrus.Fields{
+		"session":        kg.Session,
+		"parties_joined": partiesJoined,
+	}).Info("Session started")
 
-	_, err := coordinator.ExecuteKeyGeneration(coordinator.KeygenInput{
-		Server:    relayServer,
-		Key:       kg.Key,
-		Parties:   kg.Parties,
-		Session:   kg.Session,
-		ChainCode: kg.ChainCode,
+	if err != nil {
+		if err == context.DeadlineExceeded {
```

Comment:
we are not doing anything special for `context.DeadlineExceeded` error , why not just return a wrapped error